wandb/sdk/artifacts/artifact.py

```diff
@@ -2102,6 +2160,60 @@ def _load_manifest(self, url: str) -> None:
                 dep_artifact = entry._get_referenced_artifact(self._client)
                 self._dependent_artifacts.add(dep_artifact)
 
+    @staticmethod
+    def _get_gql_artifact_fragment() -> str:
+        fields = InternalApi().server_artifact_introspection()
+        fragment = """
+            fragment ArtifactFragment on Artifact {
+                id
+                artifactSequence {
+                    project {
+                        entityName
+                        name
+                    }
+                    name
+                }
+                versionIndex
+                artifactType {
+                    name
+                }
+                description
+                metadata
+                ttlDurationSeconds
+                ttlIsInherited
+                aliases {
+                    artifactCollection {
+                        project {
+                            entityName
+                            name
+                        }
+                        name
+                    }
+                    alias
+                }
+                state
+                commitHash
+                fileCount
+                createdAt
+                updatedAt
+            }
+        """
+        if "ttlDurationSeconds" not in fields and "ttlIsInherited" not in fields:
+            return fragment.replace("ttlDurationSeconds", "").replace(
+                "ttlIsInherited", ""
+            )
+        return fragment
+
+    def _ttl_duration_seconds_to_gql(self) -> Optional[int]:
+        # Set artifact ttl value to ttl_duration_seconds if the user set a value
+        # otherwise use ttl_status to indicate the backend INHERIT(-1) or DISABLE(-2) when the TTL is None
+        # When ttl_change = None its a no op since nothing changed
+        if self._ttl_change is None:
+            return None
+        if self._ttl_change == ArtifactTTLChange.INHERITED:
+            return -1
+        return self._ttl_duration_seconds or -2
```

Comment:
changed to use `_ttl_is_inherited`. Its tricky since a lot of these values can be derived from each other so sometimes we can use variables interchangeably but you're right it makes more sense to use `_ttl_is_inherited`.